### DISCUSSION TITLE
Handling Errors in batch statements

### DISCUSSION THREAD

**Discussion by fbede** [simolus3/drift#2634] [Q&A] [answered]
Please, when doing a batch insert or upsert what happens if one row fails... Does everything in the batch fail? Or does only that particular row fail?

**Comment by simolus3** [comment, marked as answer]
Batches are atomic, so a single statement failing in the batch will revert all the other ones as well. This is by design: As batches are constructed synchronously, there's no way to react to individual failing statements. If you need a different failure strategy, you can use transactions and catch failing statements if you want the rest to continue.
